01-import_and_maxfilter.py

```diff
@@ -219,63 +233,114 @@ def run_maxwell_filter(subject, session=None):
                                            recording=config.rec,
                                            space=config.space)
 
-        # read_raw_bids automatically
-        # - populates bad channels using the BIDS channels.tsv
-        # - sets channels types according to BIDS channels.tsv `type` column
-        # - sets raw.annotations using the BIDS events.tsv
-        extra_params = dict()
-        if config.allow_maxshield:
-            extra_params['allow_maxshield'] = config.allow_maxshield
-
-        raw = read_raw_bids(bids_basename=bids_basename,
-                            bids_root=config.bids_root,
-                            extra_params=extra_params,
-                            kind=config.get_kind())
-
-        # Rename events.
-        if config.rename_events:
-            rename_events(raw=raw, subject=subject, session=session)
-
-        # XXX hack to deal with dates that fif files cannot handle
-        if config.daysback is not None:
-            raw.anonymize(daysback=config.daysback)
-
-        if config.crop is not None:
-            raw.crop(*config.crop)
-
-        raw.load_data()
-
-        if hasattr(raw, 'fix_mag_coil_types'):
-            raw.fix_mag_coil_types()
+        raw = load_data(bids_basename)
+        if run_idx == 0:
+            dev_head_t = raw.info['dev_head_t']  # Re-use in all runs.
 
+        # Auto-detect bad channels.
         if config.find_flat_channels_meg or config.find_noisy_channels_meg:
             find_bad_channels(raw=raw, subject=subject, session=session,
                               task=config.get_task(), run=run)
 
+        # Maxwell-filter experimental data.
         if config.use_maxwell_filter:
-            if run_idx == 0:  # Re-use in all subsequent runs.
-                dev_head_t = raw.info['dev_head_t']
-
-            raw_sss = apply_maxwell_filter(raw=raw, subject=subject,
-                                           session=session,
-                                           dev_head_t=dev_head_t)
+            msg = 'Applying Maxwell filter to experimental data.'
+            logger.info(gen_log_message(message=msg, step=1, subject=subject,
+                                        session=session))
+
+            # Warn if no bad channels are set before Maxwell filter
+            if not raw.info['bads']:
+                msg = '\nFound no bad channels. \n '
+                logger.warn(gen_log_message(message=msg, subject=subject,
+                                            step=1, session=session))
```

Comment:
why not warn if there is no bad channel _and_ an automatic method finds a bad channel? Otherwise you're crying wolf. How are the bad channels marked by the way? Is it on data that has SSP applied on it?